demo/query_vuln.md

```diff
@@ -0,0 +1,226 @@
+```
+
+Also, clone GUAC data, this is used as test data for this demo.
+
+```bash
+git clone https://github.com/guacsec/guac-data.git
+```
+
+The rest of the demo will assume you are in the GUAC directory
+
+```bash
+cd guac
+```
+
+## Building the GUAC binaries
+
+Build the GUAC binaries using the `make` command.
+
+```bash
+make
+```
+
+## Running the GUAC Server
+
+The GUAC server can be run in different ways. For this demo, we will use the
+`guacone gql-server` command with the `--debug` flag command, which sets up a
+GraphQL endpoint and playground, and runs an in-memory backend to store the GUAC
+graph.
+
+Run this command in a separate terminal (in the same path) and keep it running
+throughout the demo.
+
+```bash
+bin/guacone gql-server --gql-debug
+```
+
+Note: As the data is stored in-memory, whenever you restart the server, the
+graph will be empty.
+
+## Running the GUAC Visualizer
+
+TODO: LINK TO GUAC VISUALIZER SET UP
+
+## Ingesting a vulnerability SPDX SBOM
+
+For demo purposes, let's ingest a known bad SPDX SBOM that contains a bunch of
+vulnerabilities. To do this, we will use the help of the `guacone` command,
+which is an all-in-one utility that can take a collection of files and ingest
+them into the GUAC graph.
+
+In your original window, run:
+
+```bash
+bin/guacone files ../guac-data/docs/spdx/spdx_vuln.json
+```
+
+This will ingest the vulnerable SPDX SBOM into GUAC so that various insights can
+be easily queried.
+
+Once ingested you will see the following message:
+
+```bash
+{"level":"info","ts":1681821120.162612,"caller":"cmd/files.go:181","msg":"[2.158961542s] completed doc {Collector:FileCollector Source:file:///../guac-data/docs/spdx/spdx_vuln.json}"}
+{"level":"info","ts":1681821120.162633,"caller":"cmd/files.go:188","msg":"collector ended gracefully"}
+{"level":"info","ts":1681821120.1626382,"caller":"cmd/files.go:201","msg":"completed ingesting 1 documents of 1"}
+```
+
+## Drawing further insight from OSV.dev
+
+One of the benefits of GUAC is that it’s not a static database, it is constantly
+evolving and trying to find more information on the artifacts ingested. To
+demonstrate this, we will utilize one of the components of GUAC known as a
+“certifier”. The role of the certifier is to continuously run and query for
+additional information from various sources (such as osv.dev and scorecard to
+start with) and keep the information specified up-to-date within GUAC.
+
+The certifier can be run in two modes, polling (for continuous updates on the
+information) or non-polling (run once and collect the data). For this demo, we
+will run it in non-polling as we want to capture the information and utilize it
+in our query.
+
+Particularly we will be running the OSV certifier, which will query osv.dev and
+determine if the various components that make up our images have vulnerabilities
+we should be worried about.
+
+To do this, run (with -p=false specifying non-polling) :
```

Comment:
i am leaning towards pointing to the docker compose for setup, and then showing that this is done all automated for you... and that because its a polling service, it may take a while, but if you want to follow along the demo more quickly, you can do it directly through running this command.

Reply:
done.